x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/extractor/fields/ExtractedField.java

```diff
@@ -108,6 +118,11 @@ public static ExtractedField newField(String alias, String name, ExtractionMetho
             }
             return new Object[0];
         }
+
+        @Override
+        protected boolean supportsFromSource() {
+            return getExtractionMethod() == ExtractionMethod.DOC_VALUE;
```

Comment:
So, we are assuming that anything that is `FromFields` that has `ExtractionMethod.DOC_VALUE` also supports extraction via `_source`? This seems OK to me for the most part, except for `GeoPointField`, which may need to override `supportsFromSource()` to always return `false`.

I think we may want to do something similar with `TimeField`

Reply:
Good catch! Will adjust.